### DISCUSSION TITLE
RofiBeats to  Play Music from Music Folder

### DISCUSSION THREAD

**Discussion by Linxford** [JaKooLit/Hyprland-Dots#259] [Ideas]
Hi bro @JaKooLit , i have just worked on this right now, and i want toknow if it will be worth it implementing it..

i can now play music from my local Music folder using the ff code,  you can try it and give suggestions

`#!/bin/bash

# Directory for icons
iDIR="$HOME/.config/swaync/icons"

# Define menu options as an associative array
declare -A menu_options

# Populate the menu_options array with music files from the Music folder
for file in ~/Music/*.mp3; do
  filename=$(basename "$file")
  menu_options["$filename"]="$file"
done

# Function for displaying notifications
notification() {
  notify-send -u normal -i "$iDIR/music.png" "Playing now: $@"
}

# Main function
main() {
  choice=$(printf "%s\n" "${!menu_options[@]}" | rofi -i -dmenu -config ~/.config/rofi/config-rofi-Beats.rasi -p "")

  if [ -z "$choice" ]; then
    exit 1
  fi

  file="${menu_options[$choice]}"

  notification "$choice"
  
  # Play the selected music file using mpv
  mpv "$file"
}

# Check if an online music process is running and send a notification, otherwise run the main function
pkill mpv && notify-send -u low -i "$iDIR/music.png" "Online Music stopped" || main
`

**Comment by Linxford** [comment]
[RofiBeats.txt](https://github.com/JaKooLit/Hyprland-Dots/files/15285855/RofiBeats.txt)

**Comment by JaKooLit** [comment]
nice idea... 

You can either integrate on the RofiBeats.sh or create a new one....

But a choice would be nice, Online or local music, shuffle, possibly add other filetypes like .ogg or something

you can make PR so you will be credited accordingly...

**Reply by Linxford** [reply]
alright sir...thanks..

will  work on that now 😋

**Comment by Linxford** [comment]
Done!!  @JaKooLit 

Here is how it goes

![swappy-20240512-105442](https://github.com/JaKooLit/Hyprland-Dots/assets/98231918/af37006e-1053-4f2c-a8cd-08b3bfd97238)

## after an option is selected ...this comes up

![swappy-20240512-105504](https://github.com/JaKooLit/Hyprland-Dots/assets/98231918/5d212a84-4735-438a-a896-93cf6d4f543f)

**Comment by JaKooLit** [comment]
https://github.com/JaKooLit/Hyprland-Dots/pull/285

https://github.com/JaKooLit/Hyprland-Dots/commit/bbb2c6d524df11b439ab3f79b31a80b06635fda2
